lib/engine/bitvector.rb

```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+
+module Engine
+  class BitVector
+    BITS_PER_GROUP = 52 # max safe JS integer size
+
+    def initialize(graph, id)
+      @graph = graph
+      @id = id
+      @next_index = 0
+      @vector = []
+    end
+
+    def [](target)
+      check_and_allocate(target)
+      ((@vector[target.bit_group] >> target.bit_index) & 1) == 1
+    end
+
+    def []=(target, val)
+      check_and_allocate(target)
+      bit = val ? 1 : 0
+      @vector[target.bit_group] |= (bit << target.bit_index)
+    end
+
+    def delete(target)
+      check_and_allocate(target)
+      @vector[target.bit_group] = (@vector[target.bit_group] | (1 << target.bit_index)) ^ (1 << target.bit_index)
+    end
+
+    private
+
+    def check_and_allocate(target)
+      if target.walk_graph != @graph || target.walk_id != @id
+        target.walk_graph = @graph
+        target.walk_id = @id
+        target.bit_group = @next_index.div(BITS_PER_GROUP)
```

Comment:
why do you use div and not / ?

Reply:
Just more explicit that it is integer division, I guess. Is it a problem?

Reply:
I'll change it